fix: various bugs in traits (#1591)
tact-lang/tact: docs/src/content/docs/ref/core-base.mdx

  label: "OTP-006: Contract Package"
---

Every [contract](/book/contracts) in Tact implicitly [inherits](/book/contracts#traits) the `BaseTrait{:tact}` trait, which contains a number of the most useful [internal functions](/book/contracts#internal-functions) for any kind of contract, and a constant `self.storageReserve{:tact}` aimed at advanced users of Tact.

> :::note
> 
>   Before 1.6.0, `BaseTrait{:tact}` was also implicitly inherited by traits, but now you need to explicitly specify `with BaseTrait{:tact}` for your traits.

Reviewer: ```suggestion
:::tip

  Prior to 1.6.0, `BaseTrait{:tact}` was also implicitly inherited by traits, but now you must explicitly specify `with BaseTrait{:tact}` for your traits to use it.
```

A small nit — we use the `tip` badge and its purple colors for denoting version-related changes

Author: Thank you!

Author: Done
